rotkehlchen/chain/evm/accounting/structures.py

```diff
@@ -27,11 +32,28 @@ class TxAccountingTreatment(Enum):
     SWAP = 0
 
 
-class TxEventSettings(NamedTuple):
-    """Settings for an event generated by a decoder"""
-    taxable: bool
-    count_entire_amount_spend: bool
-    count_cost_basis_pnl: bool
-    method: Literal['acquisition', 'spend']
-    accounting_treatment: Optional[TxAccountingTreatment] = None
-    accountant_cb: Optional[AccountantCallback] = None
+class BaseEventSettings(Generic[T]):
+    """Settings for history base entry accounting"""
+    def __init__(
+            self,
+            taxable: bool,
+            count_entire_amount_spend: bool,
+            count_cost_basis_pnl: bool,
+            method: Literal['acquisition', 'spend'],
+            accounting_treatment: Optional[TxAccountingTreatment] = None,
+            accountant_cb: Optional[HistoryBaseEntriesAccountantCallback[T]] = None,
+    ):
+        self.taxable = taxable
+        self.count_entire_amount_spend = count_entire_amount_spend
+        self.count_cost_basis_pnl = count_cost_basis_pnl
+        self.method = method
+        self.accounting_treatment = accounting_treatment
+        self.accountant_cb = accountant_cb
+
+
+class HistoryBaseEntrySettings(BaseEventSettings[HistoryBaseEntry]):
+    ...
+
+
+class TxEventSettings(BaseEventSettings[EvmEvent]):
+    ...
```

Comment:
After messing with the types I believe to have found the reason for having two classes here since there are rules that can be applied to any event and others that can be applied only to evm events. 

Still the problem appears in the 

```
self.evm_accounting_aggregators.get_accounting_settings(self.pot) |  # type: ignore[assignment]  # noqa: E501
```
that basically breaks the type checking. What do you think of my other suggestion?